File: tasks/implementations/python2/tasks/prime_sieve.py
from tasks.compat import emit

LIMITS = {'s': 50000, 'm': 125000, 'l': 250000}


def run(size, fixtures_root):
    n = LIMITS[size]
    sieve = [True] * (n + 1)
    sieve[0:2] = [False, False]
    p = 2
    while p * p <= n:
        if sieve[p]:
            for idx in xrange(p * p, n + 1, p):
                sieve[idx] = False
        p += 1
    emit(sum(1 for flag in sieve if flag))

File: tasks/implementations/rust/tasks/prime_sieve.rs
pub fn run(size: &str, _fixtures_root: &str) {
    let n = match size { "s" => 50_000, "m" => 125_000, _ => 250_000 };
    let mut sieve = vec![true; n + 1];
    sieve[0] = false; sieve[1] = false;
    let mut p = 2usize;
    while p * p <= n {
        if sieve[p] {
            let mut i = p * p;
            while i <= n { sieve[i] = false; i += p; }
        }
        p += 1;
    }
    let count = sieve.iter().filter(|x| **x).count();
    println!("{}", count);
}
